pkg/scheduler/scheduling_queue_test.go

```diff
@@ -688,11 +692,295 @@ func TestHeadroom(t *testing.T) {
 	res, err = resources.NewResourceFromConf(map[string]string{"first": "10", "second": "2"})
 	assert.NilError(t, err, "failed to create resource")
 	headRoom = leaf1.getHeadRoom()
-	if !resources.Equals(res, headRoom) {
+	maxHeadRoom = leaf1.getMaxHeadRoom()
+	if !resources.Equals(res, headRoom) || !resources.Equals(res, maxHeadRoom) {
 		t.Errorf("leaf1 queue head room not as expected %v, got: %v (err %v)", res, headRoom, err)
 	}
 	headRoom = leaf2.getHeadRoom()
+	maxHeadRoom = leaf2.getMaxHeadRoom()
+	if !resources.Equals(res, headRoom) || !resources.Equals(res, maxHeadRoom)  {
+		t.Errorf("leaf1 queue head room not as expected %v, got: %v (err %v)", res, headRoom, err)
+	}
+}
+
+func TestMaxHeadroom(t *testing.T) {
+	// create the root: nil test
```

Comment:
I am missing tests with a maxResource set on the root queue.
You can set it on the root queue via the `qi.setMaxResource()`

Reply:
That is part of the coverage: https://github.com/yangwwei/incubator-yunikorn-core/blob/d0eeadf50a51cafaca84ae7b2e7b78bf69ee0954/pkg/scheduler/scheduling_queue_test.go#L817-L832

Reply:
These tests can be greatly simplified. We also should not mix the two headroom calculation tests.
filed follow up jira to fix this

Reply:
OK, thanks!